transport/tchannel/outbound.go

```diff
@@ -176,9 +193,28 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		return nil, err
 	}
 
-	buf := bytes.NewBuffer(make([]byte, 0, _defaultBufferSize))
-	if _, err = buf.ReadFrom(resBody); err != nil {
-		return nil, err
+	var body io.ReadCloser
+	var bodySize int
+	if reuseBuffer {
+		buffer := bufferpool.NewAutoReleaseBuffer()
+		if _, err = buffer.ReadFrom(resBody); err != nil {
+			return nil, err
+		}
```

Comment:
```suggestion
	body, bodySize, err := getResponseBody(resBody, reuseBuffer)
	if err {
		return nil, err
	}
```
Method is quite huge already, it would be nice to move this into a separate method and cut short if reuse buffer is set.